### ISSUE TITLE
Error: Cannot find module 'webpack'

### ISSUE THREAD

**Issue by vedtam** [aruntk/polymer-apollo-frontpage#2]
hello @aruntk,

Sorry for bothering, I have spent a whole day trying to get this project working but without luck. Now when I clone this repo and try to start it I get:

Warnings when installing:
```
npm WARN optional Skipping failed optional dependency /chokidar/fsevents:
npm WARN notsup Not compatible with your operating system or architecture: fsevents@1.0.15
npm WARN babel-loader@6.2.7 requires a peer of webpack@1 || ^2.1.0-beta but none was installed.
npm WARN webpack-dev-server@2.1.0-beta.11 requires a peer of webpack@^2.1.0-beta.26 but none was installed.
npm WARN webpack-dev-middleware@1.8.4 requires a peer of webpack@1 || ^2.1.0-beta but none was installed.
```

Error on Startup:

```
vedtam@vedtam-K52Jc:~/test/polimer-apollo/polymer-apollo-frontpage$ sudo npm start

> frontpage-polymer-app@ start /home/vedtam/test/polimer-apollo/polymer-apollo-frontpage
> webpack-dev-server --inline --hot --port 3000

module.js:328
    throw err;
    ^

Error: Cannot find module 'webpack'
    at Function.Module._resolveFilename (module.js:326:15)
    at Function.Module._load (module.js:277:25)
    at Module.require (module.js:354:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (/home/vedtam/test/polimer-apollo/polymer-apollo-frontpage/node_modules/webpack-dev-server/lib/Server.js:13:15)
    at Module._compile (module.js:410:26)
    at Object.Module._extensions..js (module.js:417:10)
    at Module.load (module.js:344:32)
    at Function.Module._load (module.js:301:12)
    at Module.require (module.js:354:17)

npm ERR! Linux 4.4.0-47-generic
npm ERR! argv "/usr/bin/nodejs" "/usr/bin/npm" "start"
npm ERR! node v4.2.6
npm ERR! npm  v3.5.2
npm ERR! code ELIFECYCLE
npm ERR! frontpage-polymer-app@ start: `webpack-dev-server --inline --hot --port 3000`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the frontpage-polymer-app@ start script 'webpack-dev-server --inline --hot --port 3000'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the frontpage-polymer-app package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     webpack-dev-server --inline --hot --port 3000
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs frontpage-polymer-app
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls frontpage-polymer-app
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /home/vedtam/test/polimer-apollo/polymer-apollo-frontpage/npm-debug.log

```

**Comment by vedtam** [comment]
The way I could fix this was via (adding back webpack and removing the ^ -sign from webpack-dev-server):

```
"webpack": "2.1.0-beta.22",
 "webpack-dev-server": "2.1.0-beta.0"
```

But now with this fixed, in my polymer app, wile the GraphiQl server is running I get:

```
Uncaught TypeError: Cannot convert undefined or null to object
    at Function.keys (<anonymous>)
    at DollarApollo.process (eval at <anonymous> (http://localhost:3000/dist/build.js:1455:1), <anonymous>:353:38)
    at HTMLElement.beforeRegister (eval at <anonymous> (http://localhost:3000/dist/build.js:1455:1), <anonymous>:418:26)
    at HTMLElement._invokeBehavior (http://localhost:3000/src/vulcanized.html:462:4)
    at HTMLElement._doBehavior (http://localhost:3000/src/vulcanized.html:455:6)
    at HTMLElement.registerCallback (http://localhost:3000/src/vulcanized.html:205:6)
    at desugar (http://localhost:3000/src/vulcanized.html:96:11)
    at window.Polymer (http://localhost:3000/src/vulcanized.html:80:15)
    at eval (eval at <anonymous> (http://localhost:3000/dist/build.js:1286:1), <anonymous>:38:1)
    at Object.<anonymous> (http://localhost:3000/dist/build.js:1286:1)
```

**Comment by aruntk** [comment]
@vedtam Check now. I've fixed the bug.

**Comment by vedtam** [comment]
@aruntk  Yeeyyy! :)

Please could you elaborate what the problem was? I was digging into it all day yesterday and this morning, I must know.

Was it something related to the behaviour?

**Comment by aruntk** [comment]
There were two problems
- webpack and webpack-dev-server peer-dependency mismatch. 

``` json
 // package.json
  "webpack": "2.1.0-beta.22",
  "webpack-dev-server": "2.1.0-beta.0"
```

 Earlier it was 

``` json
 // package.json
  "webpack": "2.1.0-beta.22",
 "webpack-dev-server": "^2.1.0-beta.0"
```

 Which means webpack-dev-server version will be the latest (which was 2.1.0-beta.0 when I created the repo) and webpack version stuck at `2.1.0-beta.22`. They have created some breaking changes in webpack since then. 
- At [this line](https://github.com/aruntk/polymer-apollo/blob/master/src/polymer-apollo.js#L242) it is expecting options.variables to be an object. I've now added a condition to avoid undefined.
